main.rb

```diff
@@ -1,95 +1,22 @@
 load "bst.rb"
+load "linked_list.rb"
 
-tree = Bst.new
+different_data_structure = {bst: 0, linked_list: 1}
 
-puts "0 : Quit"
-puts "1 : Insert no"
-puts "2 : Largest element"
-puts "3 : Smallest element"
-puts "4 : Inorder traversal"
-puts "5 : Postorder traversal"
-puts "6 : Preorder traversal"
-puts "7 : Search element"
-puts "8 : Delete Element"
-puts "9 : All Path from Root to leaf"
-puts "10: Input from file"
+puts	"#{different_data_structure.fetch(:bst)}: Binary search tree" ,
+	"#{different_data_structure.fetch(:linked_list)}: Linked List  "
 
-different_task = 
-		{quit: 0,
-		 insert_no: 1,
-		 largest_element: 2,
-		 smallest_element: 3,
-     inorder_traversal: 4,
-     postorder_traversal: 5,
-     preorder_traversal: 6,
-     search_element: 7,
-     delete_node: 8,
-     all_path_root_leaf: 9,
-     input_file: 10 }
+print "enter your choice : "
+type_of_ds = gets
 
-loop do
+case type_of_ds.to_i
```

Comment:
what if user gives bad input such as
```
"abc"
"1a"
```

Reply:
not just here but in any place where user is giving inputs, what happens if user gives wrong inputs?